Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Plan.hs

```diff
@@ -324,7 +324,7 @@ getSubsriptionConfigAndPlanSubscription ::
 getSubsriptionConfigAndPlanSubscription serviceName (driverId, _, merchantOpCityId) mbDPlan = do
   subscriptionConfig <- CQSC.findSubscriptionConfigsByMerchantOpCityIdAndServiceName merchantOpCityId serviceName >>= fromMaybeM (NoSubscriptionConfigForService merchantOpCityId.getId $ show serviceName)
   vehicleCategory <- getVehicleCategory driverId subscriptionConfig
-  let isVehicleChanged = (mbDPlan >>= (.vehicleCategory)) /= vehicleCategory && isJust mbDPlan
+  let isVehicleChanged = (mbDPlan >>= (.vehicleCategory)) /= Just vehicleCategory && isJust mbDPlan
```

Comment:
We need to make sure than vehicleCategory is backfilled in driverPlan else this check will fail.

Reply:
`getCategoryFromPlanOrSubscriptionConfig` in transformer takes care of it.